data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/LogstashPluginMapper.java

```diff
@@ -24,18 +26,19 @@
 class LogstashPluginMapper {
     private final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
     private final AttributesMapperProvider attributesMapperProvider;
+    private final CustomPluginMapperCreator creator;
```

Comment:
I'm not sure I follow how this will solve the current problem. The `CustomPluginMapper` maps from a Logstash plugin into multiple Data Prepper plugins. It is not mapping from attributes to a plugins.

I'm definitely interested in better names. I originally proposed `CustomPluginMapper` because `LogstashPluginMapper` was already taken. But, this class is package private and could be renamed. The `CustomPluginMapper` is what other libraries would use and so we should be sure we end up on the right name for that.

Also, I agree that we should combine `CustomPluginMapperCreator` and `AttributesMapperCreator` into one class. But, I think that having this code as a separate class is good. It separates concerns. And it simplifies the unit testing and code of `LogstashPluginMapper`. They are both package protected and not for external use.

Ultimately, I'd rather put complexity into this package if it simplifies plugin authors' work of writing custom plugin or attributes mapping. So I'd rather we base our decision on what makes a simpler interface to implement for custom mapping.

Reply:
@dlvenable, thanks for your feedback. I don't think we should have a large class design debate this in a PR. All of this should have happen before we start coding the converter but here we are :) . I can put together an issue to debate this class design. I don't want to hold up this work anymore